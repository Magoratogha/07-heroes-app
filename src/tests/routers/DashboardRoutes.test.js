import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/authContext";
import DashboardRoutes from "../../routers/DashboardRoutes";

describe('Test on <DashboardRoutes />', () => {

    const contextValue = {
        user: {
            name: 'Test',
            logged: true
        }
    };

    it('should render without crashing', () => {
        const { container } = render(<AuthContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={ ['/'] }>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>);
        expect(container).toMatchSnapshot();
        expect(screen.getByText(contextValue.user.name, { selector: 'span' })).toBeInTheDocument();
        expect(screen.getByText('Marvel Screen', { selector: 'h1' })).toBeInTheDocument();
    });

    it('should render DC path', () => {
        const { container } = render(<AuthContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={ ['/dc'] }>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>);
        expect(container).toMatchSnapshot();
        expect(screen.getByText('DC Screen', { selector: 'h1' })).toBeInTheDocument();
    });
});