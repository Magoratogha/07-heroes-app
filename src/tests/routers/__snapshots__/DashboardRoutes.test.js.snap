// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <DashboardRoutes /> should render DC path 1`] = `
<div>
  <nav
    class="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/marvel"
      >
        HeroesApp
      </a>
      <div
        class="navbar-collapse"
      >
        <div
          class="navbar-nav"
        >
          <a
            class="nav-item nav-link"
            href="/marvel"
          >
            Marvel
          </a>
          <a
            aria-current="page"
            class="nav-item nav-link active"
            href="/dc"
          >
            DC
          </a>
          <a
            class="nav-item nav-link"
            href="/search"
          >
            Search
          </a>
        </div>
      </div>
      <div
        class="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end "
      >
        <ul
          class="navbar-nav ml-auto"
        >
          <span
            class="nav-item nav-link text-warning"
          >
            Test
          </span>
          <button
            class="btn nav-item nav-link"
          >
            Logout
          </button>
        </ul>
      </div>
    </div>
  </nav>
  <div
    class="container mt-4 mb-5"
  >
    <h1>
      DC Screen
    </h1>
    <hr />
    <div
      class="row row-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
    >
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Batman"
                class="card-img"
                src="/assets/images/dc-batman.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Batman
                </h5>
                <p
                  class="card-text"
                >
                  Bruce Wayne
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    Detective Comics #27
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/dc-batman"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Superman"
                class="card-img"
                src="/assets/images/dc-superman.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Superman
                </h5>
                <p
                  class="card-text"
                >
                  Kal-El
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    Action Comics #1
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/dc-superman"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Flash"
                class="card-img"
                src="/assets/images/dc-flash.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Flash
                </h5>
                <p
                  class="card-text"
                >
                  Jay Garrick
                </p>
                <p
                  class="text-muted"
                >
                  Jay Garrick, Barry Allen, Wally West, Bart Allen
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    Flash Comics #1
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/dc-flash"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Green Lantern"
                class="card-img"
                src="/assets/images/dc-green.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Green Lantern
                </h5>
                <p
                  class="card-text"
                >
                  Alan Scott
                </p>
                <p
                  class="text-muted"
                >
                  Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    All-American Comics #16
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/dc-green"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Green Arrow"
                class="card-img"
                src="/assets/images/dc-arrow.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Green Arrow
                </h5>
                <p
                  class="card-text"
                >
                  Oliver Queen
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    More Fun Comics #73
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/dc-arrow"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Wonder Woman"
                class="card-img"
                src="/assets/images/dc-wonder.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Wonder Woman
                </h5>
                <p
                  class="card-text"
                >
                  Princess Diana
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    All Star Comics #8
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/dc-wonder"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Martian Manhunter"
                class="card-img"
                src="/assets/images/dc-martian.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Martian Manhunter
                </h5>
                <p
                  class="card-text"
                >
                  J'onn J'onzz
                </p>
                <p
                  class="text-muted"
                >
                  Martian Manhunter
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    Detective Comics #225
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/dc-martian"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Robin/Nightwing"
                class="card-img"
                src="/assets/images/dc-robin.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Robin/Nightwing
                </h5>
                <p
                  class="card-text"
                >
                  Dick Grayson
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    Detective Comics #38
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/dc-robin"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Blue Beetle"
                class="card-img"
                src="/assets/images/dc-blue.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Blue Beetle
                </h5>
                <p
                  class="card-text"
                >
                  Dan Garret
                </p>
                <p
                  class="text-muted"
                >
                  Dan Garret, Ted Kord, Jaime Reyes
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    Mystery Men Comics #1
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/dc-blue"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Black Canary"
                class="card-img"
                src="/assets/images/dc-black.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Black Canary
                </h5>
                <p
                  class="card-text"
                >
                  Dinah Drake
                </p>
                <p
                  class="text-muted"
                >
                  Dinah Drake, Dinah Lance
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    Flash Comics #86
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/dc-black"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test on <DashboardRoutes /> should render without crashing 1`] = `
<div>
  <nav
    class="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/marvel"
      >
        HeroesApp
      </a>
      <div
        class="navbar-collapse"
      >
        <div
          class="navbar-nav"
        >
          <a
            class="nav-item nav-link"
            href="/marvel"
          >
            Marvel
          </a>
          <a
            class="nav-item nav-link"
            href="/dc"
          >
            DC
          </a>
          <a
            class="nav-item nav-link"
            href="/search"
          >
            Search
          </a>
        </div>
      </div>
      <div
        class="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end "
      >
        <ul
          class="navbar-nav ml-auto"
        >
          <span
            class="nav-item nav-link text-warning"
          >
            Test
          </span>
          <button
            class="btn nav-item nav-link"
          >
            Logout
          </button>
        </ul>
      </div>
    </div>
  </nav>
  <div
    class="container mt-4 mb-5"
  >
    <h1>
      Marvel Screen
    </h1>
    <hr />
    <div
      class="row row-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
    >
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Spider Man"
                class="card-img"
                src="/assets/images/marvel-spider.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Spider Man
                </h5>
                <p
                  class="card-text"
                >
                  Peter Parker
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    Amazing Fantasy #15
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/marvel-spider"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Captain America"
                class="card-img"
                src="/assets/images/marvel-captain.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Captain America
                </h5>
                <p
                  class="card-text"
                >
                  Steve Rogers
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    Captain America Comics #1
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/marvel-captain"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Iron Man"
                class="card-img"
                src="/assets/images/marvel-iron.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Iron Man
                </h5>
                <p
                  class="card-text"
                >
                  Tony Stark
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    Tales of Suspense #39
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/marvel-iron"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Thor"
                class="card-img"
                src="/assets/images/marvel-thor.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Thor
                </h5>
                <p
                  class="card-text"
                >
                  Thor Odinson
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    Journey into Myster #83
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/marvel-thor"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Hulk"
                class="card-img"
                src="/assets/images/marvel-hulk.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Hulk
                </h5>
                <p
                  class="card-text"
                >
                  Bruce Banner
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    The Incredible Hulk #1
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/marvel-hulk"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Wolverine"
                class="card-img"
                src="/assets/images/marvel-wolverine.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Wolverine
                </h5>
                <p
                  class="card-text"
                >
                  James Howlett
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    The Incredible Hulk #180
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/marvel-wolverine"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Daredevil"
                class="card-img"
                src="/assets/images/marvel-daredevil.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Daredevil
                </h5>
                <p
                  class="card-text"
                >
                  Matthew Michael Murdock
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    Daredevil #1
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/marvel-daredevil"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Hawkeye"
                class="card-img"
                src="/assets/images/marvel-hawkeye.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Hawkeye
                </h5>
                <p
                  class="card-text"
                >
                  Clinton Francis Barton
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    Tales of Suspense #57
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/marvel-hawkeye"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Cyclops"
                class="card-img"
                src="/assets/images/marvel-cyclops.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Cyclops
                </h5>
                <p
                  class="card-text"
                >
                  Scott Summers
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    X-Men #1
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/marvel-cyclops"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col mb-2 animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-5 py-3 ps-4 pe-0"
            >
              <img
                alt="Silver Surfer"
                class="card-img"
                src="/assets/images/marvel-silver.jpg"
              />
            </div>
            <div
              class="col-7"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Silver Surfer
                </h5>
                <p
                  class="card-text"
                >
                  Norrin Radd
                </p>
                <p
                  class="card-text"
                >
                  <small>
                    The Fantastic Four #48
                  </small>
                </p>
                <a
                  class="btn btn-outline-dark btn-sm"
                  href="/hero/marvel-silver"
                >
                  View more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
